builtin/tfc/config_sourcer_test.go

```diff
@@ -58,33 +61,160 @@ func TestConfigSourcer(t *testing.T) {
+		v := result[0]
 
-	require.Equal("HOSTNAME", v.Name)
-	ve, ok := v.Result.(*pb.ConfigSource_Value_Error)
-	if ok {
-		require.False(ok, "error reading value: %s", ve.Error.Message)
-	}
+		require.Equal("outputs", includes)
 
-	require.Equal(
-		"arn:aws:iam::797645259670:role/waypoint-circleci-release-pipeline",
-		v.Result.(*pb.ConfigSource_Value_Value).Value,
-	)
+		require.Equal("HOSTNAME", v.Name)
+		ve, ok := v.Result.(*pb.ConfigSource_Value_Error)
+		if ok {
+			require.False(ok, "error reading value: %s", ve.Error.Message)
+		}
+
+		resultJson, ok := v.Result.(*pb.ConfigSource_Value_Json)
+		require.True(ok)
+
+		var resultMarshalled []string
+		require.NoError(json.Unmarshal(resultJson.Json, &resultMarshalled))
+
+		require.True(ok)
+		require.ElementsMatch([]string{
+			"subnet-03afefcc38a919083",
+			"subnet-087ea2efd4f009fc5",
+			"subnet-09e02553a59746ba1",
+			"subnet-05a2d1112fbca071c",
+			"subnet-0d5f650b4d0eebc56",
+		}, resultMarshalled)
+	})
+
+	t.Run("Read all outputs", func(t *testing.T) {
+
+		cases := []struct {
+			tfcRespJsonTestdataPath string
+			numOutputs              int
+			spotChecks              map[string]interface{}
+		}{
+			{
+				"testdata/state-version.json",
+				3,
+				map[string]interface{}{
+					"alb_listener_arn": "arn:aws:elasticloadbalancing:us-east-1:797645259670:listener/app/acmeapp1-dev/0ed92920e20ed1dc/07e51901e3cec498",
+				},
+			},
+			{
+				"testdata/state-version-complex.json",
+				14,
+				map[string]interface{}{
+					"region": "us-east-1",
+					"ecs_task_subnets": []string{
+						"subnet-03afefcc38a919083",
+						"subnet-087ea2efd4f009fc5",
+						"subnet-09e02553a59746ba1",
+						"subnet-05a2d1112fbca071c",
+						"subnet-0d5f650b4d0eebc56",
+					},
+				},
+			},
+		}
+
+		for _, tt := range cases {
+			tfcRespJsonTestdataPath = tt.tfcRespJsonTestdataPath
+
+			result, err := cs.read(ctx, log, []*component.ConfigRequest{
+				{
+					Name: "HOSTNAME",
+					Config: map[string]string{
+						"organization": "foocorp",
+						"workspace":    "databases",
+						"all_outputs":  "true",
+					},
+				},
+			})
+			require.NoError(err)
+			require.NotNil(result)
+			require.Len(result, 1)
+
+			v := result[0]
+
+			ve, ok := v.Result.(*pb.ConfigSource_Value_Error)
+			if ok {
+				require.False(ok, "error reading value: %s", ve.Error.Message)
+			}
+
+			jsonResult, ok := v.Result.(*pb.ConfigSource_Value_Json)
+			require.True(ok)
+
+			require.NotEmpty(jsonResult.Json)
+
+			var jsonResults map[string]interface{}
+			require.NoError(json.Unmarshal(jsonResult.Json, &jsonResults))
+			require.Len(jsonResults, tt.numOutputs)
+			for k, v := range tt.spotChecks {
+				actualV, ok := jsonResults[k]
+				require.True(ok)
+
+				switch actualV.(type) {
+				case string:
+					require.Equal(v, actualV)
+				case []interface{}:
+					require.ElementsMatch(v, actualV)
+				}
+			}
+		}
+
+	})
+
+	t.Run("Can read a single var with a complex type", func(t *testing.T) {
+
```

Comment:
Missing test?

Reply:
Ah, good catch! I actually have that case covered earlier, so this can be deleted.